samples/19.c.single-sign-on-for-teams-apps/web/public/index.html

```diff
@@ -0,0 +1,163 @@
+        background-repeat: no-repeat;
+        background-size: auto 50%;
+        margin: 0;
+      }
+
+      #webchat {
+        border-radius: 4px;
+        box-shadow: 0 0 8px rgba(0, 0, 0, .08);
+        height: calc(100% - 20px);
+        left: 10px;
+        overflow: hidden;
+        position: fixed;
+        top: 10px;
+        width: 360px;
+      }
+    </style>
+  </head>
+  <body>
+    <noscript>You need to enable JavaScript to run this app.</noscript>
+    <script>
+      // Settings.
+      const OAUTH_REDIRECT_URI = '/api/oauth/authorize';
+
+      // This is a helper function for fetching JSON resources.
+      async function fetchJSON(url, options = {}) {
+        const res = await fetch(
+          url,
+          {
+            ...options,
+            headers: {
+              ...options.headers,
+              accept: 'application/json'
+            }
+          }
+        );
+
+        if (!res.ok) {
+            throw new Error(`Failed to fetch JSON due to ${ res.status }`);
+        }
+
+        return await res.json();
+      }
+
+      // Verify if the access token is valid or not.
+      async function verifyAccessToken(accessToken) {
+        const res = await fetch(
+          "https://graph.microsoft.com/v1.0/me",
+          {
+            headers: {
+              authorization: `Bearer ${accessToken}`
+            }
+          }
+        );
+
+        return res.ok;
+      }
+
+      // This will request Microsoft Teams to start a popup dialog for sign-in process.
+      function authorize() {
+        return new Promise((resolve, reject) => {
+          window.microsoftTeams.authentication.authenticate({
+            failureCallback: reject,
+            height: 535,
+            successCallback: ({ accessToken }) => resolve(accessToken),
+            url: window.location.origin + '/api/oauth/authorize',
+            width: 600
+          });
+        });
+      }
+
+      // This will request Microsoft Teams to send context information to the web page.
+      // Microsoft Teams suggest that the context information should not be trusted and should only used as hint for further authentication.
+      function getTeamsContext() {
+        return new Promise(resolve => window.microsoftTeams.getContext(resolve));
+      }
+
+      (async function main() {
+        // Initialize Microsoft Teams JavaScript SDK
+        window.microsoftTeams.initialize();
+
+        // When OAuth provider performs callback, the access token will be saved into session storage and redirected back to this page. See /web/src/routes/oauth/callback.js.
+        // We will retrive the token from session storage from here.
+        let oauthAccessToken = window.localStorage.getItem('OAUTH_ACCESS_TOKEN');
+
+        // Get the Microsoft Teams context.
+        const teamsContext = await getTeamsContext();
+
+        // If we are not authenticated, or the valid is not working, we will redirect the end-user to the sign-in page.
```

Comment:
validation is not working?

Reply:
Fixed.

"or the access token is not working"